docs/DeveloperGuide.md

```diff
@@ -1,11 +1,233 @@
+---
+layout: default
+title: Developer Guide
+---
+* Table of Contents
+  {:toc}
+
+--------------------------------------------------------------------------------------------------------------------
+
 # Developer Guide
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+References
+1. Developer Guide: https://se-education.org/addressbook-level3/DeveloperGuide.html 
+2. User Guide: https://se-education.org/addressbook-level3/UserGuide.html
+3. 
+
 ## Design & implementation
 
+## Design
+
+### Architecture
+
+<img src="images/ArchitectureDiagram.png" width="280" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+### UI component
+
+### Logic component
+
+### Model component
+
+### Storage component
+
+In this application, it uses text files to store all data, i.e, recipes and ingredients. 
+
+
+When booting up the application, `restoreSavedData` will be called to get both recipes and ingredients 
+from the previous session.
+
+<img src="images/RestoreIngredientStorageSequenceDiagram.png" width="347" />
+<img src="images/RestoreRecipeStorageSequenceDiagram.png" width="347" />
+
+When a user exits the application, `saveData` will be called and it will convert all data from recipes and
+ingredients into a string which will be put in their own respective text files.
+
+<img src="images/StoreIngredientStorageSequenceDiagram.png" width="525" />
+<img src="images/StoreRecipeStorageSequenceDiagram.png" width="706" />
+
+
+## Implementation
+### Help Feature
+The help feature is facilitated by the `HelpCommand` class. By calling `executeCommand` on the class, it will invoke the `Ui` class to print the user help commands.
+
+
+- **Step 1**
+
+  Input will be sent from the main `EssenMakanan` class to the `Parser` to identify the command type
+
+
+- **Step 2**
+
+  A new `HelpCommand` object will be created and will be sent back to main
+
+
+- **Step 3**
+
+  `commandObject#executeCommand()` will be called which in turn calls `Ui#showCommands()`
+
+
+- **Step 4**
+
+  Finally `Ui#showCommands()` will call `Ui#showRecipeCommands()`, `Ui#showIngredientCommands()`, `Ui#showOtherCommands()` to print all commands for recipe, ingredient and others respectively
+
+
+<img src="images/HelpFunctionSequenceDiagram.png" width="732" />
```

Comment:
Perhaps the command constructor could end here?
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/110757597/64496332-8826-46ec-bc2d-4cc2d181ca79)